GHCN.py

```diff
@@ -0,0 +1,1044 @@
+    # loop over linear derived variables/extremes
+    if any(var.linear for var in self.extremes): print('\n computing (linear) monthly variables:')
+    for var in self.extremes:      
+      varobj = self.dataset[var.name] # get variable object
+      wrfvar = ravmap[var.name]
+      if var.linear:
+        # compute from available monthly data
+        print("   {:<15s} {:s}".format(var.name,str(tuple(self.varmap.get(varname,varname) for varname in var.prerequisites))))
+        monlytmp = var.computeValues(monlydata, aggax=1, delta=86400.)
+        monlydata[wrfvar] = monlytmp
+      tmpload = monlydata[wrfvar]
+      assert varobj.shape == tmpload.shape
+      varobj.load(tmpload)
+    # determine actual length of records (valid data points)
+    minlen = None 
+    for varname in self.variables.iterkeys():
+      rec_len = self.dataset['stn_'+varname+'_len'] # get variable object
+      varobj = self.dataset[varname]
+      stlen,tlen = varobj.shape
+      assert stlen == rec_len.shape[0]
+      tmp = tlen - np.isnan(varobj.getArray(unmask=True, fillValue=np.NaN)).sum(axis=1)
+      tmp = np.asanyarray(tmp, dtype=rec_len.dtype)
+      assert tmp.shape == rec_len.shape
+      rec_len.load(tmp) 
+      if minlen is None: minlen = tmp
+      else: minlen = np.minimum(minlen,tmp)
+    # save minimum as overall record length
+    self.dataset['stn_rec_len'].load(minlen)
+    # synchronize data, i.e. write to disk
+    self.dataset.sync()
+    
+    
+## load pre-processed GHCN station time-series
+def loadGHCN_TS(name=None, filetype=None, varlist=None, varatts=None, 
+              filelist=None, folder=None, **kwargs): 
+  ''' Load a monthly time-series of pre-processed GHCN station data. '''
+  if filetype is None: raise ArgumentError, "A 'filetype' needs to be specified ('temp' or 'precip','all')."
+  elif not filetype in ('temp','precip','all'): raise ArgumentError
+  name = name or 'GHCN' # 'ec{:s}'.format(filetype) # prepend ec to the filetype
+  #if prov is not None and not isinstance(prov,basestring): raise TypeError
+  if folder is None: folder = avgfolder
+  if filelist is None: filelist = [tsfile.format(filetype)]
+   # if prov: filelist = [tsfile_prov.format(filetype, prov)]
+    #else: filelist = [tsfile.format(filetype)]
+  if varlist is not None: # translate varlist
+    varlist = translateVarNames(varlist, varatts)
+  # open NetCDF file (name, varlist, and varatts are passed on directly)
+  dataset = DatasetNetCDF(name=name, folder=folder, filelist=filelist, varlist=varlist, varatts=varatts, 
+                            multifile=False, ncformat='NETCDF4', **kwargs)
+  return dataset
+# wrapper
+def loadGHCN_StnTS(name=None, station=None, varlist=None, varatts=varatts, **kwargs):
+  ''' Load a monthly time-series of pre-processed GHCN station data. '''
+  if station is None: raise ArgumentError, "A 'filetype' needs to be specified ('ghcntemp' or 'ghcnprecip')."
+  elif station in ('ghcntemp','ghcnprecip'):

```

Comment:
You are right, I will change that.

发自我的 iPhone

> 在 2016年7月24日，15:25，"Andre R. Erler" <notifications@github.com> 写道：
> 
> In GHCN.py:
> 
> > +  #if prov is not None and not isinstance(prov,basestring): raise TypeError
> > +  if folder is None: folder = avgfolder
> > +  if filelist is None: filelist = [tsfile.format(filetype)]
> > +   # if prov: filelist = [tsfile_prov.format(filetype, prov)]
> > +    #else: filelist = [tsfile.format(filetype)]
> > +  if varlist is not None: # translate varlist
> > +    varlist = translateVarNames(varlist, varatts)
> > +  # open NetCDF file (name, varlist, and varatts are passed on directly)
> > +  dataset = DatasetNetCDF(name=name, folder=folder, filelist=filelist, varlist=varlist, varatts=varatts, 
> > +                            multifile=False, ncformat='NETCDF4', **kwargs)
> > +  return dataset
> > +# wrapper
> > +def loadGHCN_StnTS(name=None, station=None, varlist=None, varatts=varatts, **kwargs):
> > +  ''' Load a monthly time-series of pre-processed GHCN station data. '''
> > +  if station is None: raise ArgumentError, "A 'filetype' needs to be specified ('ghcntemp' or 'ghcnprecip')."
> > +  elif station in ('ghcntemp','ghcnprecip'):
> I guess you need to add 'all' here as well.
> 
> ―
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>